### ISSUE TITLE
Migrate to https

### ISSUE THREAD

**Issue by HarelM** [IsraelHikingMap/Site#259]
Not sure where the change should be made, either iis or web.config.

**Comment by zstadler** [comment]
@HarelM 
Please clarify the issue, as https is supported.
What are the requested changes?
What bugs are there?

**Comment by HarelM** [comment]
Auto redirect from http to https for all users. Will solve the gps bug.

**Comment by valleyofdawn** [comment]
Only when hitting the track button,  or always?

**Comment by HarelM** [comment]
Always.

**Comment by zstadler** [comment]
Automatic redirection will cause everyone to loose their saved layers.

I don't think this is a good trade-off for the GPS limitation.
- The GPS tracking user scenario is less important
- It is primarily applicable to mobile devices
- On mobile devices the maps are primarily used within applications.

I can think of several options to address the "locate me" limitation:

1. Redirect to https only when the "locate me" button is pushed, on browsers that need https, and after user acknowledgment that tracks, POIs, and use maps will be lost.
2. Temporarily save map contents, including hidden layers and OSM authentication, to the server, and restore that saved state from a new https connection. 
3. Disable (either grey-out or hide) the "locate me" button on http connections from browsers that require https.

**Comment by HarelM** [comment]
I don't think a lot of users will care about losing their layers as I don't think a lot of users has many custom layer, I know you and guy have but I don't think it's common.
I do however think that migrating to https now will probably make things easier in the future - like the GPS bug (I don't know what chrome is planning but this seems like a roadmap to make the browser more secure and I prefer to fix it now instead of later).
The GPS bug is more severe than keeping the layers IMHO.

1. No only mobile devices has GPS, tablets too.
Redirect on GPS activation will still cause layer loss.
2. I think it will be very hard as the data is stored in cookies and not in the server.
maybe we can change it as part of #258 - so that you'll have your layer every where as long as you are signed in to OSM - this won't prevent layer loss out-of-the-box and I prefer layer loss over hard to maintain code.
3. Users will not know how to activate it, it doesn't really solve anything besides the horrible error message...

**Comment by zstadler** [comment]
We seem to disagree on relative importance of the GPS bug vs. users layers.
Perhaps we can agree on using option 2 above...

The technical approach I have for option 2 is to leverage the server's "map sharing" mechanism to store the user's map, and then redirect the browser to the https address of a sharing link which will restore the user's map.

The differences between the existing "map sharing" mechanism and the proposed "http to https map migration" mechanism would be:
- An "http to https" link will also save and restore hidden layers
- An "http to https" link will be removed from the server once it is accessed, or after some predefined time since its creation.

**Comment by HarelM** [comment]
We shouldn't abuse the sharing for this, there are easier ways. Let's try and focus on option 2 as part of OSM integration.

**Comment by zstadler** [comment]
Interestingly, there is a [similar GitHub issue](/openstreetmap/operations/issues/117) under discussion in the [openstreetmap.org operations](/openstreetmap/operations) project

**Comment by HarelM** [comment]
TL;DR, Bottom line is that they too feel that the show my location is important enough.

**Comment by HarelM** [comment]
This is an extremely patchy solution to the GPS issue - new behavior is to redirect the user when he clicks the GPS button to the https site so that when he tried again it will work :-(.
I don't think this is a good solution though...

**Comment by HarelM** [comment]
We were not able to agree on the subject of redirecting http to https, @zstadler does not agree with me that we should force the users to always use https.
Until we visit this topic again I'll keep the code commented out.
Closing...

**Comment by HarelM** [comment]
Custom layers moved to server, no objection to move to Https.
